himl/main.py

```diff
@@ -21,6 +21,31 @@ def run(self, args):
         opts = parser.parse_args(args)
         self.do_run(opts)
 
+    """
+    example of custom merge strategy. pass this function to ConfigProcessor to override default merging behavior for list data type, 
+    which is append
+    """
+    def strategy_merge_override(self,config, path, base, nxt):
+        """merge list of objects. if objects have same id, nxt replaces base."""
+        """if remove flag is present in nxt item, remove base and not add nxt"""
+        result = deepcopy(base)
+        for nxto in nxt:
+            for baseo in result:
+                # if list is a list of dicts, check for 'id' field for uniqueness  
+                if 'id' in baseo and 'id' in nxto and baseo['id'] == nxto['id']:
+                    result.remove(baseo) #same id, remove previous item
+                    continue
+                # if list is a list of strings, simply compare string values, and don't duplicate strings
+                # this behavior is actually the same as the default append_unique (using set) strategy 
+                # but unfortuntely deepmerge does not provide chained executed strategies, only fallback strategies
+                if isinstance(baseo,str) and isinstance(nxto,str) and baseo == nxto:
```

Comment:
this should be on line 35 - otherwise a list like:
```
myList:
 - id1
 - id2
 - id3
```

would crash in flames ;)

Reply:
Good catch. Will update. Should we leave this custom function in himl package? maybe as an example to showcase how to build a custom merging strategy? Will check in to tie everything together.